utils/navigator_robotx_comms/nodes/navigator_robotx_comms.py

```diff
@@ -0,0 +1,413 @@
+#!/usr/bin/env python
+
+"""
+RobotX Communications: A node that handles communications
+with the RobotX Technical Director network.
+"""
+
+import datetime
+import math
+import socket
+import time
+import threading
+
+import rospy
+import tf.transformations as trans
+from geometry_msgs.msg import PointStamped
+from mil_tools import rosmsg_to_numpy
+from nav_msgs.msg import Odometry
+from navigator_msgs.srv import *
+from mil_tools import thread_lock
+
+lock = threading.Lock()
+
+rospy.init_node("robotx_comms")
+
+
+class RobotXHeartbeatMessage():
+    """
+    Handles formation and sending of heartbeat message.
+    """
+
+    def __init__(self):
+
+        self.team_id = rospy.get_param('team_id')
+        self.message_id = "RXHRB"
+        self.gps_array = None
+        self.odom = None
+        self.auv_status = 1
+        self.system_mode = 2
+
+        rospy.Subscriber("lla", PointStamped, self.gps_coord_callback)
+        rospy.Subscriber("odom", Odometry, self.gps_odom_callback)
+        # rospy.Subscriber("auv_status", String, self.auv_status_callback)
```

Comment:
As this is the class for the message, it should not contain subscribers, just serialize/serialize the message. Put these in the client class